### DISCUSSION TITLE
How to use Metrics (if possible)?

### DISCUSSION THREAD

**Discussion by x-xyzxyz** [stayallive/wp-sentry#193] [Q&A] [answered]
Hello, and thanks as always for your amazing work with this package!

Is it possible to send custom JS Metrics to Sentry? 

https://docs.sentry.io/platforms/javascript/metrics/

I couldn't find anything in your docs, discussions and issues.

Thank you!
Alessandro

**Comment by stayallive** [comment]
The JS SDK is loaded so you can just call something like this from the docs:

```
Sentry.metrics.increment("button_click", 1, {
  tags: { browser: "Firefox", app_version: "1.0.0" },
});
```

Keep in mind that this is JavaScript code and you cannot emit JavaScript metrics from PHP code.

**Reply by x-xyzxyz** [reply]
Thanks, I did try the exact same snippet before posting:

```js
jQuery(document).ready(function ($) {
	$('.button').click(function () {
		Sentry.metrics.increment('test_click', 1, {
			tags: { browser: 'Firefox', app_version: '1.0.0' },
		});
		console.log('test_click');
	});
});
```

but even though the console shows `test_click` when I click a button, I don't see the event in the Metrics section on Sentry, so I thought I'd have to rely on some of your php hooks. I know I have to wait a few minutes before they show up, but it's been an hour now and still nothing.

`wp-sentry-integration@7.15.0-scoped:php@8.2.18:sdk@4.7.0:js-sdk@7.116.0`

![CleanShotX - Arc  13 06 2024 at 12-49-01 @2x](https://github.com/stayallive/wp-sentry/assets/47320294/a12f830a-88e0-4917-9f18-7b587a89962e)



Not sure if I'm perhaps looking in the wrong place, or doing something wrong. Everything else is working fine so I'm pretty sure to be set up correctly!

**Reply by stayallive** [reply, marked as answer]
Ah I see the issue... I read over it. Apparently v7 of the SDK need an extra integration for metrics to function. We should probably add this by default or have an easier way to enable this. But for now you can paste this snippet in your theme functions.php for example.

```
add_action( 'wp_enqueue_scripts', function () {
	wp_add_inline_script( 'wp-sentry-browser', 'function wp_sentry_hook(options) { options.integrations.push(Sentry.metrics.metricsAggregatorIntegration()); }', 'before' );
} );
```

After this metrics should start working!

**Reply by x-xyzxyz** [reply]
bless you @stayallive that worked 😄
Thank you so much!
